langchain/src/runnables/remote.ts

```diff
@@ -325,4 +335,73 @@ export class RemoteRunnable<
     });
     return IterableReadableStream.fromReadableStream(stream);
   }
+
+  async *streamLog(
+    input: RunInput,
+    options?: Partial<CallOptions>,
+    streamOptions?: Omit<LogStreamCallbackHandlerInput, "autoClose">
+  ): AsyncGenerator<RunLogPatch> {
+    const [config, kwargs] =
+      this._separateRunnableConfigFromCallOptions(options);
+    const stream = new LogStreamCallbackHandler({
+      ...streamOptions,
+      autoClose: false,
+    });
+    const { callbacks } = config;
+    if (callbacks === undefined) {
+      config.callbacks = [stream];
+    } else if (Array.isArray(callbacks)) {
+      config.callbacks = callbacks.concat([stream]);
+    } else {
+      const copiedCallbacks = callbacks.copy();
+      copiedCallbacks.inheritableHandlers.push(stream);
+      config.callbacks = copiedCallbacks;
+    }
+    const getExtraOptionsFromInput = () => {
+      const optionKeys = [
+        "diff",
+        "include_names",
+        "include_types",
+        "include_tags",
+        "exclude_names",
+        "exclude_types",
+        "exclude_tags",
+      ];
+      const extraOptions: Record<string, unknown> = {};
+      for (const key of optionKeys) {
+        const keyAsKeyof = key as keyof RunInput;
+        if (input[keyAsKeyof] !== undefined && input[keyAsKeyof] !== null) {
+          extraOptions[key] = input[keyAsKeyof];
+        }
+      }
+      return extraOptions;
+    };
+    const response = await this.post<{
+      input: RunInput;
+      config?: RunnableConfig;
+      kwargs?: Omit<Partial<CallOptions>, keyof BaseCallbackConfig>;
+      diff?: boolean;
+      include_names?: Array<string>;
+      include_types?: Array<string>;
+      include_tags?: Array<string>;
+      exclude_names?: Array<string>;
+      exclude_types?: Array<string>;
+      exclude_tags?: Array<string>;
+    }>("/stream_log", {
+      input,
+      config,
+      kwargs,
+      ...getExtraOptionsFromInput(),
+    });
+    const { body } = response;
+    if (!body) {
+      throw new Error(
+        "Could not begin remote stream log. Please check the given URL and try again."
+      );
+    }
+    const runnableStream = convertEventStreamToIterableReadableDataStream(body);
```

Comment:
not super familiar here - will `body` be the ReadableStream that this function expects?

Should confirm the objects yielded here are the same as from the langserve python client. I think this might be missing some of the `jsonpatch` mechanics

Reply:
Yes it should. The `body` object is of type `ReadableStream<Uint8Array> | null`. Not 100% on the `jsonpatch` though, could you provide some context as to what you're looking for?

Reply:
https://github.com/langchain-ai/langserve/blob/main/langserve/client.py#L673-L677

It's possible this is implemented somewhere else. Can test tomorrow if helpful, and just want to confirm this yields equivalent objects as LangServe does, and not just the patch ops

https://jsonpatch.com/

Reply:
@efriis just pushed a change which should satisfy this, lmk thoughts